File: src/test/java/org/example/AuthorRepositoryTest.java
package org.example;

import org.junit.Test;

import java.util.List;

import static org.junit.Assert.assertEquals;

public class AuthorRepositoryTest extends TestcontainersTestcode {


    @Test
    public void save_should_set_id() throws Exception {
        withConnection(connection -> {
            AuthorRepository sut = new AuthorRepository(connection);
            Author author = new Author("Kevin");

            sut.save(author);
            assertEquals(Long.valueOf(1), author.getId());
        });
    }

    @Test
    public void findAll_should_contain_saved_author() throws Exception {
        withConnection(connection -> {
            AuthorRepository sut = new AuthorRepository(connection);

            Author author = new Author("Michael");
            sut.save(author);

            List<Author> authors = sut.findAll();
            assertEquals(1, authors.size());
            assertEquals("Michael", authors.get(0).getName());
        });
    }

}

File: src/test/kotlin/AuthorRepositoryTest.kt
import org.example.Author
import org.example.AuthorRepository
import org.junit.Assert
import org.junit.Test

class AuthorRepositoryTest : TestcontainersTestcode() {
    @Test
    @Throws(Exception::class)
    fun save_should_set_id() {
        withConnection { connection ->
            val sut = AuthorRepository(connection)
            val author = Author(name = "Kevin")
            val savedAuthor = sut.save(author)
            Assert.assertEquals(java.lang.Long.valueOf(1), savedAuthor.id)
        }
    }

    @Test
    @Throws(Exception::class)
    fun findAll_should_contain_saved_author() {
        withConnection { connection ->
            val sut = AuthorRepository(connection)
            val author = Author(name = "Michael")
            sut.save(author)
            val authors = sut.findAll()
            Assert.assertEquals(1, authors.size.toLong())
            Assert.assertEquals("Michael", authors[0].name)
        }
    }
}